x/gov/client/rest/query.go

```diff
@@ -296,10 +296,11 @@ func queryVoteHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
 		}
 
 		var vote types.Vote
-		if err := cliCtx.Codec.UnmarshalJSON(res, &vote); err != nil {
```

Comment:
we should instead check if the `res` is empty for the case when the vote is pruned (i.e use a `switch` statement) and leave this as well:

```go
var vote types.Vote
err := cliCtx.Codec.UnmarshalJSON(res, &vote)

switch {
    case len(res) == 0:
        // vote has been prunned; skip error
    case err != nil:
        rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
        return
}
```

Reply:
@fedekunze If I try to build from `master` branch, it seems like `make build` fails due to not having `cosmos-sdk/x/genaccounts/client/cli`. Is `master` branch of Cosmos SDK the most up-to-date branch? What version should I use to build Gaia with it?

Reply:
@fedekunze I tested with the case when the vote is pruned, but it seems like the problem persists.